angular/src/app/modules/review/review-detail/review-detail.component.ts

```diff
@@ -857,13 +866,172 @@ export class ReviewDetailComponent extends PagedListingComponentBase<ReviewDetai
+  }
+  
+  sortByReviewer(){
+    if(this.sortStatus){
+      this.listReviewIntern = _.orderBy(this.listReviewIntern, ['reviewerName'], ['asc']);
+    }
+    else {
+      this.listReviewIntern = _.orderBy(this.listReviewIntern, ['reviewerName'], ['desc']);
+    }
+    this.sortStatus = !this.sortStatus;
+  }
+
+ 
+  sortByStatus(){
+    if(this.sortStatus){
+      this.listReviewIntern = _.orderBy(this.listReviewIntern, ['status'], ['asc']);
+    }
+    else {
+      this.listReviewIntern = _.orderBy(this.listReviewIntern, ['status'], ['desc']);
+    }
+    this.sortStatus = !this.sortStatus;
+  }
+
+  setAll(completed: boolean) {
+    this.allComplete = completed;
+    if (this.listReviewIntern == null) {
+      return;
+    }
+
+    this.listSelectedItem = []; 
+
+    this.listReviewIntern.forEach((reviewDetail) => {
+      if (reviewDetail.status === 4) {
+        reviewDetail.selected = completed;
+
+        if (completed) {
+          this.listSelectedItem.push({
+            selected: completed,
+            id: reviewDetail.id,
+            status: reviewDetail.status,
+          });
+        }
+      }
+    });
+    this.checkSelectedCheckbox();
+  }
+
+  ishandleSelectlistReviewInternItem(index, $event) {
+    console.log("item", this.listReviewIntern[index]);
+    if (index != undefined && this.listReviewIntern[index]) {
+      this.listReviewIntern[index].selected = $event.checked;
+  
+      if (!this.listSelectedItem) {
+        this.listSelectedItem = [];
+      }
+  
+      const item = this.listSelectedItem.find(
+        (item) => item.id == this.listReviewIntern[index].id
+      );
+  
+      if (item) {
+        item.selected = $event.checked;
+      } else {
+        this.listSelectedItem.push({
+          selected: $event.checked,
+          id: this.listReviewIntern[index].id,
+          status: this.listReviewIntern[index].status
+        });
+      }
+  
+      this.updateAllComplete();
+      this.checkSelectedCheckbox();
+    }
+  }
+
+  checkSelectedCheckbox() {
+    this.selectedCheckboxCount = this.listSelectedItem.filter(
+      (x) => x.selected
+    ).length;
+  }
+
+  updateAllComplete() {
+    this.allComplete =
+      this.listReviewIntern != null &&
+      this.listReviewIntern.every((t) => this.getIsSelected(t));
+  }
+
+  
+  getIsSelected(item) {
+    return this.listSelectedItem.find((c) => c.id == item.id)
+      ? this.listSelectedItem.find((c) => c.id == item.id).selected
+      : item.selected;
+  }
+
+  someComplete(): boolean {
+    if (this.listReviewIntern == null || this.listReviewIntern.length == 0) {
+      return false;
+    }
+  
+    return this.listReviewIntern.some((t) => this.getIsSelected(t)) && !this.allComplete;
+  }
+
+  hasSelectedItems(){
+    return this.listReviewIntern.some(item => item.selected === true);
+  }
+
+  headPmAprroveOrRejectAll(status: number){
+    if(this.selectedCheckboxCount === 0){
+      abp.notify.error("Need to select at least one record");
```

Comment:
Should change the message to `Please select list record before performing this action` or disabled button if list selected is empty.